modules/ml/src/tree.cpp

```diff
@@ -48,6 +48,47 @@ namespace ml {
 
 using std::vector;
 
+struct ParallelBestSplit
+{
+    ParallelBestSplit (const vector<int>& sidx, DTreesImpl * dtree)
+            : tree(dtree), _sidx(sidx)
```

Comment:
`_sidx` is used as read-only data in the parallel body. There is no need to make copy, you can use the original one for all threads.

Reply:
@asmorkalov good point - made the member variable a const reference as well.